stimpacks/deploy/config.go

```diff
@@ -0,0 +1,352 @@
+package deploy
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/PremiereGlobal/stim/pkg/utils"
+	v2e "github.com/PremiereGlobal/vault-to-envs/pkg/vaulttoenvs"
+)
+
+const (
+	defaultContainerRepo   = "premiereglobal/kube-vault-deploy"
+	defaultContainerTag    = "0.3.1"
+	defaultDeployDirectory = "./"
+	defaultDeployScript    = "deploy.sh"
+	defaultConfigFile      = "./stim.deploy.yaml"
+)
+
+// Config is the root structure for the deployment configuration
+type Config struct {
+	configFilePath string
+	Deployment     Deployment     `yaml:"deployment"`
+	Global         Global         `yaml:"global"`
+	Environments   []*Environment `yaml:"environments"`
+	environmentMap map[string]int
+}
+
+// Deployment describes details about the deployment assets (directories, files, etc)
+type Deployment struct {
+	Directory         string    `yaml:"directory"`
+	Script            string    `yaml:"script"`
+	Container         Container `yaml:"container"`
+	fullDirectoryPath string
+}
+
+// Container describes the container used for Docker deployments
+type Container struct {
+	Repo string `yaml:"repo"`
+	Tag  string `yaml:"tag"`
+}
+
+// Global describes global environment specs
+type Global struct {
+	Spec *Spec `yaml:"spec"`
+}
+
+// Spec contains the spec of a given environment/instance
+type Spec struct {
+	Kubernetes      Kubernetes        `yaml:"kubernetes"`
+	Secrets         []*v2e.SecretItem `yaml:"secrets"`
+	EnvironmentVars []*EnvironmentVar `yaml:"env"`
+}
+
+// Kubernetes describes the Kubernetes configuration to use
+type Kubernetes struct {
+	ServiceAccount string `yaml:"serviceAccount"`
+	Cluster        string `yaml:"cluster"`
+}
+
+// Environment describes a deployment environment (i.e. dev, stage, prod, etc.)
+type Environment struct {
+	Name        string      `yaml:"name"`
+	Spec        *Spec       `yaml:"spec"`
+	Instances   []*Instance `yaml:"instances"`
+	instanceMap map[string]int
+}
+
+// Instance describes an instance of a deployment within an environment (i.e. us-west-2 for env prod)
+type Instance struct {
+	Name string `yaml:"name"`
+	Spec *Spec  `yaml:"spec"`
+}
+
+// EnvironmentVar describes a shell env var to be injected into the deployment environment
+type EnvironmentVar struct {
+	Name  string `yaml:"name"`
+	Value string `yaml:"value"`
+}
+
+// parseConfig opens the deployment config file and ensures it is valid
+func (d *Deploy) parseConfig() {
+
+	d.config = Config{}
+
+	configFile := d.stim.GetConfig("deploy.file")
+
+	if configFile == "" {
+		setConfigDefault(&configFile, defaultConfigFile)
```

Comment:
This seems like a double check here, you can pretty much just do:
```
configFile = defaultConfigFile 
```
Since you know its not set at this point, no reason to pass pointers around and check len of the string again

Reply:
Yeah, I could.  Was just trying to stick to the pattern elsewhere for setting default values.